android/engine/src/main/java/org/smartregister/fhircore/engine/rulesengine/RulesFactory.kt

```diff
@@ -387,12 +387,19 @@ constructor(
         ?: emptyList()
     }
 
-    /** This function combines all string indexes to comma separated */
+    /** This function combines all the string indexes to a comma separated list */
     fun joinToString(source: MutableList<String?>): String {
-      source.removeIf { it == null }
-      val inputString = source.joinToString()
-      val regex = "(?<=^|,)[\\s,]*(\\w[\\w\\s]*)(?=[\\s,]*$|,)".toRegex()
-      return regex.findAll(inputString).joinToString(", ") { it.groupValues[1] }
+      return joinToString(source, DEFAULT_REGEX, DEFAULT_STRING_SEPARATOR)
+    }
+
+    /**
+     * This function combines all string indexes to a list separated by the separator defined by the
+     * implementor
+     */
+    fun joinToString(sourceString: MutableList<String?>, regex: String, separator: String): String {
+      sourceString.removeIf { it == null }
+      val inputString = sourceString.joinToString()
+      return regex.toRegex().findAll(inputString).joinToString(separator) { it.groupValues[1] }
```

Comment:
@ellykits I decide to overload the method so that the default regex can still be used if a project has no need in providing a new regex